expression/aggregation/base_func.go

```diff
@@ -104,6 +104,8 @@ func (a *baseFuncDesc) TypeInfer(ctx sessionctx.Context) error {
 		a.typeInfer4MaxMin(ctx)
 	case ast.AggFuncBitAnd, ast.AggFuncBitOr, ast.AggFuncBitXor:
 		a.typeInfer4BitFuncs(ctx)
+		// make cast to int explicit
+		a.Args[0] = expression.WrapWithCastAsInt(ctx, a.Args[0])
```

Comment:
Can we make this implicit cast into function `typeInfer4BitFuncs`? Like https://github.com/solotzg/tidb/blob/eeb23695b8fb913c4a3d92f3528ed9b7f43f537d/expression/aggregation/base_func.go#L273 does?